File: assertion-roulette/python-unittest/CrudAppServiceTests_Should_Create_Async.py
import unittest

class TestCrudAppService(unittest.TestCase):
    def test_should_create(self):
        user_output = _product_crud_app_service.create(CreateProductInput(code="create_async_product_code", name="Create Async Product Name"))
        default_test_db_context.save_changes()
        another_scope_db_context = get_default_test_db_context()
        inserted_product_dto = another_scope_db_context.products.find(user_output.id)
        self.assertIsNotNone(user_output)
        self.assertIsNotNone(inserted_product_dto)
        self.assertEqual(user_output.code, inserted_product_dto.code)

File: assertion-roulette/java-junit/CrudAppServiceTests_Should_Create_Async.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class CrudAppServiceTests {

    @Test
    public void shouldCreateAsync() throws Exception {
        var userOutput = _productCrudAppService.createAsync(new CreateProductInput("create_async_product_code", "Create Async Product Name"));
        defaultTestDbContext.saveChanges();
        var anotherScopeDbContext = getDefaultTestDbContext();
        var insertedProductDto = anotherScopeDbContext.getProducts().findById(userOutput.getId());
        assertNotNull(userOutput);
        assertNotNull(insertedProductDto);
        assertEquals(userOutput.getCode(), insertedProductDto.getCode());
    }
}
